tools/memap.py

```diff
@@ -486,13 +495,13 @@ def main():
 
     parser.add_argument('file', help='memory map file')
 
-    parser.add_argument('-t', '--toolchain', dest='toolchain', help='select a toolchain used to build the memory map file (ARM, GCC_ARM, IAR)',\
-                        required=True)
+    parser.add_argument('-t', '--toolchain', dest='toolchain', help='select a toolchain used to build the memory map file (%s)' % ", ".join(MemapParser.toolchains),\
```

Comment:
The way it is printing now is what toolchains the parse function expects. If this should be changed, that can be done in another PR where the toolchains array is replaced (L461) along with the code that handles the dispatch to the parsers (L474-483).